qiime_studio/server.py

```diff
@@ -13,6 +13,14 @@
 studio.after_request(add_cors_headers)
 
 
+# No output is visible, as for some reason it is held back
+# by the server flushed at seemingly random intervals
+# also, what should be the stdout, is being sent through stderr??
+def flush_stdio():
+    sys.stdout.flush()
+    sys.stderr.flush()
+
```

Comment:
It actually has to do with the fact that when Python detects it is using a shared pipe, it buffers it's output (which isn't what we really want), but when it is in terminal, it always flushes.